### PR TITLE
check the required fields of polaris when the config mode starts

### PR DESCRIPTION
**Pull request by mazengrun**
feature: check the required fields of polaris when the config mode starts  #1439

### CODE REVIEW

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java [hunk 1]
```diff
@@ -144,6 +144,33 @@ public BeforeCheckConfiguration.BeforeCheck dynamicThreadPoolBeforeCheckBean(@Au
                         }
                     }
 
+                    Map<String, String> polaris = properties.getPolaris();
+                    if (MapUtil.isNotEmpty(polaris)) {
+                        String namespace = polaris.get("namespace");
```
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java:149]: Don't use magic variables

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java [hunk 2]
```diff
@@ -144,6 +144,33 @@ public BeforeCheckConfiguration.BeforeCheck dynamicThreadPoolBeforeCheckBean(@Au
...
+                                    "Web server failed to start. The dynamic thread pool polaris namespace is empty.",
+                                    "Please check whether the [spring.dynamic.thread-pool.polaris.namespace] configuration is empty or an empty string.");
+                        }
+                        String fileGroup = polaris.get("file.group");
+                        if (StringUtil.isBlank(fileGroup)) {
+                            throw new ConfigEmptyException(
+                                    "Web server failed to start. The dynamic thread pool polaris file group is empty.",
+                                    "Please check whether the [spring.dynamic.thread-pool.polaris.file.group] configuration is empty or an empty string.");
+                        }
+                        String fileName = polaris.get("file.name");
```
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java:161]: Don't use magic variables

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java [hunk 3]
```diff
@@ -144,6 +144,33 @@ public BeforeCheckConfiguration.BeforeCheck dynamicThreadPoolBeforeCheckBean(@Au
...
+                        if (StringUtil.isBlank(fileName)) {
+                            throw new ConfigEmptyException(
+                                    "Web server failed to start. The dynamic thread pool polaris file name is empty.",
+                                    "Please check whether the [spring.dynamic.thread-pool.polaris.file.name] configuration is empty or an empty string.");
+                        }
+                        String fileType = polaris.get("file.type");
```
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java:167]: Don't use magic variables

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java [hunk 4]
```diff
@@ -144,6 +144,33 @@ public BeforeCheckConfiguration.BeforeCheck dynamicThreadPoolBeforeCheckBean(@Au
...
 
+                    Map<String, String> polaris = properties.getPolaris();
+                    if (MapUtil.isNotEmpty(polaris)) {
+                        String namespace = polaris.get("namespace");
+                        if (StringUtil.isBlank(namespace)) {
+                            throw new ConfigEmptyException(
+                                    "Web server failed to start. The dynamic thread pool polaris namespace is empty.",
+                                    "Please check whether the [spring.dynamic.thread-pool.polaris.namespace] configuration is empty or an empty string.");
+                        }
+                        String fileGroup = polaris.get("file.group");
```
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java:155]: Duplicate code maybe we can encapsulate in a method to make the code more concise.

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java (hunk 1 above)
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: Don't use magic variables

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java (hunk 4 above)
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: Don't use magic variables

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java (hunk 2 above)
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: Don't use magic variables

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java (hunk 3 above)
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: Don't use magic variables

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java (hunk 1 above)
Comment (mazengrun) [reply threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: Thank you very much! I am so sorry this is my first time to participate in open source projects, and I reresearched and modified the code after reading your comment.

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java [hunk 5]
```diff
@@ -144,6 +144,40 @@ public BeforeCheckConfiguration.BeforeCheck dynamicThreadPoolBeforeCheckBean(@Au
                         }
                     }
 
+                    Map<String, Object> polaris = properties.getPolaris();
+                    if (MapUtil.isNotEmpty(polaris)) {
+                        String namespace = polaris.get("namespace").toString();
```
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java:150]: Maybe namespace we define a static constant is more appropriate

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java [hunk 6]
```diff
@@ -144,6 +144,40 @@ public BeforeCheckConfiguration.BeforeCheck dynamicThreadPoolBeforeCheckBean(@Au
...
+                        if (StringUtil.isBlank(namespace)) {
+                            throw new ConfigEmptyException(
+                                    "Web server maybe fail to start. The dynamic thread pool polaris namespace is empty.",
+                                    "Please check whether the [spring.dynamic.thread-pool.polaris.namespace] configuration is empty or an empty string.");
+                        }
+                        if (polaris.get("file") instanceof Map) {
+                            Map<String, String> polarisFile = (Map<String, String>)polaris.get("file");
```
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: Maybe file we define a static constant is more appropriate

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java [hunk 7]
```diff
@@ -144,6 +144,40 @@ public BeforeCheckConfiguration.BeforeCheck dynamicThreadPoolBeforeCheckBean(@Au
...
+                            String fileGroup = polarisFile.get("group");
```
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: Maybe "group" we define a static constant is more appropriate

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java [hunk 8]
```diff
@@ -144,6 +144,40 @@ public BeforeCheckConfiguration.BeforeCheck dynamicThreadPoolBeforeCheckBean(@Au
...
+                                        "Web server maybe fail to start. The dynamic thread pool polaris file group is empty.",
+                                        "Please check whether the [spring.dynamic.thread-pool.polaris.file.group] configuration is empty or an empty string.");
+                            }
+                            String fileName = polarisFile.get("name");
+                            if (StringUtil.isBlank(fileName)) {
+                                throw new ConfigEmptyException(
+                                        "Web server maybe fail to start. The dynamic thread pool polaris file name is empty.",
+                                        "Please check whether the [spring.dynamic.thread-pool.polaris.file.name] configuration is empty or an empty string.");
+                            }
+                            String fileType = polarisFile.get("type");
```
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: Maybe "type" we define a static constant is more appropriate

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java [hunk 9]
```diff
@@ -39,6 +40,26 @@ public class BeforeCheckConfiguration {
     private final String bootstrapPropertiesClassName = "cn.hippo4j.springboot.starter.config.BootstrapProperties";
     private final String bootstrapConfigPropertiesClassName = "cn.hippo4j.threadpool.dynamic.mode.config.properties.BootstrapConfigProperties";
 
+    private static final String POLARIS_NAMESPACE = "${spring.dynamic.thread-pool.polaris.namespace}";
```
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: The same configuration and constants exist in `PolarisRefresherHandler`, so maybe it's easier to maintain the code without having to maintain two copies of the same configuration.

Comment (mazengrun) [reply threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: Thanks for your guidance, but I have a question that when the configuration file contains the field and the value is the default value, how to distinguish whether the default value is used?

Comment (weihubeats) [reply threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java:43]: So I think if you add a preemptive null judgement isn't it possible to remove the default values

Comment (weihubeats) [reply threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: So I think if you add a preemptive null judgement isn't it possible to remove the default values

**Comment by weihubeats** [comment]
We also understand the wrong issues, the checks here are only for the ConfigurationProperties injected properties, and @value has nothing to do, so we actually do not have to consider the complexity, just need to `BootstrapPropertiesInterface` in the properties of the check can be

File: threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java [hunk 10]
```diff
@@ -144,6 +145,40 @@ public BeforeCheckConfiguration.BeforeCheck dynamicThreadPoolBeforeCheckBean(@Au
...
+                                        "Please check whether the [spring.dynamic.thread-pool.polaris.file.group] configuration is empty or an empty string.");
+                            }
+                            String fileName = polarisFile.get("name");
+                            if (StringUtil.isBlank(fileName)) {
+                                throw new ConfigEmptyException(
+                                        "Web server maybe fail to start. The dynamic thread pool polaris file name is empty.",
+                                        "Please check whether the [spring.dynamic.thread-pool.polaris.file.name] configuration is empty or an empty string.");
+                            }
+                            String fileType = polarisFile.get("type");
+                            if (StringUtil.isBlank(fileType)) {
```
Comment (weihubeats) [inline threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: If or else throws an exception?

Comment (mazengrun) [reply threadpool/core/src/main/java/cn/hippo4j/core/enable/BeforeCheckConfiguration.java]: Thanks to the teacher for criticizing and correcting me. I promise I will not make such low-level mistakes again in the future！

**Review by weihubeats — APPROVED** [review]
LGTM

**Pull request merged**